src/libraries/Microsoft.PowerFx.Interpreter/Functions/Library.cs

```diff
@@ -32,8 +32,28 @@ private static T GetService<T>(this IServiceProvider services, T defaultService)
 
         public static IEnumerable<TexlFunction> FunctionList => FuncsByName.Keys;
 
+        private static readonly object LockObject = new object();
+
+        public static readonly IReadOnlyDictionary<TexlFunction, AsyncFunctionPtr> FuncsByName;
```

Comment:
nit: Funcs by name isn't quite right, the key here isn't the name it's the full function object

Reply:
True - this is what we had before, I didn't want to change more than necessary... I can update it, though

Reply:
Actually, it's a public field, so it's possible that someone else is using it; I won't update it (at least not in this PR)

Reply:
It's safe - 
public field on *internal* class ... 
And in Interpreter, so it's not included by PAClient (and products don't have internals access to interpreter)

Reply:
Updated the function name